airflow/utils/db.py

```diff
@@ -83,6 +83,23 @@ def merge_conn(conn, session=None):
         session.commit()
 
 
+@event.listens_for(settings.engine, "connect")
```

Comment:
It basically checks when a new connecting is made. You cannot use the same db connection in a fork so we force the db logic to make a new connection when it detects that the pid of the process that created the connection is not the same as it is currently in.